sql/src/main/java/io/crate/execution/IncrementalPageBucketReceiver.java

```diff
@@ -92,14 +96,22 @@ public void setBucket(int bucketIdx, Bucket rows, boolean isLast, PageResultList
         // We make sure only one accumulation operation runs at a time because the state is not thread-safe.
         synchronized (accumulateRowsFunction) {
             if (currentlyAccumulating == null) {
-                currentlyAccumulating = CompletableFuture.supplyAsync(() -> accumulateRowsFunction.apply(rows), executor);
+                try {
+                    currentlyAccumulating = CompletableFuture.supplyAsync(() -> accumulateRowsFunction.apply(rows), executor);
```

Comment:
What do you think about having a function like this:

```
    public static <U> CompletableFuture<U> supplyAsync(Supplier<U> supplier, Executor executor) {
        try {
            return CompletableFuture.supplyAsync(supplier, executor);
        } catch (EsRejectedExecutionException | RejectedExecutionException e) {
            return failedFuture(e);
        }
    }
```

Maybe within our `CompletableFutures` class and then make use of the forbidden APIs stuff to forbid using the bare `supplyAsync` API?

Reply:
I wasn't aware we're using that _already_. Yes, that's a good shout, I'll  change this now.